tests/integration/release/parallel_test_launcher.py

```diff
@@ -0,0 +1,218 @@
+
+        # Remove completed jobs from the remaining jobs set
+        remaining_jobs -= set(completed_jobs)
+
+        if remaining_jobs:
+            print(
+                f"{len(remaining_jobs)} jobs have not finished. Checking again in {poll_interval // 60} minutes..."
+            )
+            time.sleep(poll_interval)  # Wait for the specified interval before checking again
+
+
+def parse_outputs(output_dir: str, job_ids: list[str] | None = None) -> None:
+    """Parses the output files to see how many jobs failed, didn't complete, and succeeded
+    in the specified directory, prints this to screen, and writes a summary CSV with this info.
+
+    Args:
+        output_dir (str): Directory where Slurm output/error files are stored.
+        job_ids (list[str] | None): List of Slurm job IDs to check. If provided, will verify that all expected outputs are present.
+    """
+    job_info = []
+
+    output_files = glob.glob(f"{output_dir}/*.out")
+    if job_ids:
+        if len(output_files) < len(job_ids):
+            print(
+                f"Warning: Expected {len(job_ids)} job outputs, but found only {len(output_files)} in {output_dir}."
+            )
+            print(
+                "This may indicate that some outputs are still being written. Rerun parse_outputs after all outputs have been written"
+            )
+            return
+        else:
+            print("All expected output files found.")
+
+    # iterate through all files/jobs
+    for file in output_files:
+        with open(file, "r") as f:
+            lines = f.readlines()
+            if not lines:
+                continue  # Skip empty files
+            last_line = lines[-1].strip()
+
+        if not (last_line.startswith("=") and last_line.endswith("=")):
+            outcome = "not_completed"
+        elif "failed" in last_line.lower():
+            outcome = "failed"
+        else:
+            outcome = "passed"
+        job_info.append({"outcome": outcome, "filepath": str(file)})
+
+    job_info_df = pd.DataFrame(job_info).sort_values(by="outcome")
+    job_info_df.to_csv(f"{output_dir}/summary_results.csv", index=False)
+
+    num_failures = sum(job_info_df["outcome"] == "failed")
+    num_incomplete = sum(job_info_df["outcome"] == "not_completed")
+    if num_failures:
+        print(
+            f"FAIL: {num_failures} tests failed and {num_incomplete} tests did not finish. See {output_dir}/summary_results.csv for details."
+        )
+    else:
+        print("All submitted tests passed successfully.")
+
+
+if __name__ == "__main__":
+    csv_file = "data/parameters.csv"
+    timestamp = datetime.now().strftime("%d-%H-%M-%S")
+    output_dir = f"/mnt/team/simulation_science/priv/engineering/pseudopeople_release_testing/logs/{timestamp}"  # Directory where Slurm output/error files are stored
+    job_ids = []
+    submission_failures = []
+
+    # Step 1: Read the CSV and generate and run Slurm scripts
+    with open(csv_file, mode="r") as file:
```

Comment:
I don't love that you're doing lots of real work, including the actual job submissions themselves, within the context manager. Can you just pd.read_csv into memory and work with that instead of having the file open the whole time?